### PR TITLE
Update vcpkg create template

### PR DESCRIPTION
**Pull request by vpaladino778**
Pull request in response to issue #8363 

Changes to the template that `vcpkg create` creates

- [x] Added example `vcpkg_fail_port_install
- [x] Removed `include(vcpkg_common_functions)`
- [x] Added hint to see `vcpkg_common_definitions` for more variables
- [x] Added `vcpkg_fixup_cmake_targets` example to template
- [x] Added `vcpkg_check_features` example to template

Let me know if anyone can think of any other additions that would be helpful. 
I left the new additions commented out because i don't believe they would apply to every port file (Except maybe vcpkg_fixup_cmake_targets maybe?)

### CODE REVIEW

**Comment by Voskrese** [comment]
maybe it's worth replacing?

`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/@PORT@ RENAME copyright)`

to

`configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)`

**Comment by vpaladino778** [comment]
> maybe it's worth replacing?
> 
> `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/@PORT@ RENAME copyright)`
> 
> to
> 
> `configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)`

What benefit would there be using file-install vs configure_file-COPYONLY?

**Comment by Voskrese** [comment]
Favor? No. But it is logical to use a copy instead of installation. Well, the additional message will be deleted in the terminal.

I'm not insisting, it's just a suggestion :wink:

**Comment by vpaladino778** [comment]
@Voskrese No worries. I'm sorry if i came off as rude, I was just genuinely curious as to the differences.  I think functionally they are the same, and i believe the `file(install)` syntax might be a bit more clear to someone who isn't as familiar with CMake. So i think i'll just leave it as is for the moment.

File: scripts/templates/portfile.in.cmake [hunk 1]
```diff
@@ -30,6 +32,14 @@ vcpkg_extract_source_archive_ex(
     #   002_more_port_fixes.patch
 )
 
+# # Check if one or more features are a part of a package installation.
+# vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
```
Comment (vicroms) [inline scripts/templates/portfile.in.cmake]: Perhaps add a link to `vcpkg_check_features` documentation like you did for `vcpkg_common_definitions` and `vcpkg_fixup_cmake_targets`.

Comment (vpaladino778) [reply scripts/templates/portfile.in.cmake]: Done

**Review by vicroms — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**